evm/vm/forks/byzantium/opcodes.py

```diff
@@ -0,0 +1,126 @@
+import copy
+import functools
+
+from cytoolz import merge
+
+from evm import constants
+from evm.exceptions import (
+    WriteProtection,
+)
+from evm import opcode_values
+from evm import mnemonics
+
+from evm.opcode import as_opcode
+
+from evm.logic import (
+    call,
+    context,
+    logging,
+    storage,
+    system,
+)
+
+from evm.vm.forks.spurious_dragon.opcodes import SPURIOUS_DRAGON_OPCODES
+
+
+def no_static(opcode_fn):
+    @functools.wraps(opcode_fn)
+    def inner(computation):
+        if computation.msg.is_static:
+            raise WriteProtection("Cannot modify state while inside of a STATICCALL context")
```

Comment:
Also, the exception msg here is quite misleading, as it's saying we cannot modify the state when the msg is actually static? Or have I gotten this backwards?

Reply:
I think the txt is correct.

> "Since we are inside of a STATICCALL context, the state cannot be modified